### PR TITLE
feat: support sorted position delete writer

### PR DESCRIPTION
**Pull request by ZENOTME**
This PR mainly to complete position delete writer and add integrate test:
1. support sorted positioned delete writer 
2. move writer to  file_writer module 
3. refine location generator to support delete file 
4. refine partition splitter (I find that this is not quite relate others. But it's hard to seperate it out.🥵 Hope so will not make review too complicate, if you feel necessary to seperate it, feel free to tell me)
5. support writer builder interface to expose sorted positione delete writer 
6. integrate test for posistion delete

### CODE REVIEW

File: icelake/src/io/file_writer/position_delete_writer.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+/// A PositionDeleteWriter used to write position delete, it will sort the incoming delete by file_path and pos.
+pub struct SortedPositionDeleteWriter {
+    operator: Operator,
+    table_location: String,
+    location_generator: Arc<FileLocationGenerator>,
+    table_config: TableConfigRef,
+
+    delete_cache: BTreeMap<String, Vec<i64>>,
+    record_num: usize,
+    max_record_num: usize,
```
Comment (liurenjie1024) [inline icelake/src/io/file_writer/position_delete_writer.rs:25]: Put this in `TableConfig`?

File: icelake/src/io/file_writer/position_delete_writer.rs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+use super::rolling_writer::RollingWriter;
+
+/// A PositionDeleteWriter used to write position delete, it will sort the incoming delete by file_path and pos.
+pub struct SortedPositionDeleteWriter {
+    operator: Operator,
+    table_location: String,
+    location_generator: Arc<FileLocationGenerator>,
+    table_config: TableConfigRef,
+
+    delete_cache: BTreeMap<String, Vec<i64>>,
```
Comment (liurenjie1024) [inline icelake/src/io/file_writer/position_delete_writer.rs:23]: Java version uses roaring bitmap, which saves a lot of memory, how about we also use it?

File: icelake/src/io/position_delete_writer.rs
Comment (liurenjie1024) [inline icelake/src/io/position_delete_writer.rs:1]: Remove this file?

File: icelake/tests/delete_test.rs [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+    PositionDeleteTest::new(docker_compose, poetry, catalog)
+}
+
+fn main() {
+    // Parse command line arguments
+    let args = Arguments::from_args();
+
+    // let catalogs = vec!["storage", "rest"];
+    let catalogs = vec!["storage"];
```
Comment (liurenjie1024) [inline icelake/tests/delete_test.rs:298]: Use line above?

File: icelake/tests/delete_test.rs [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+    let mut tests = Vec::with_capacity(2);
+    for catalog in &catalogs {
+        let fixture = create_test_fixture("delete_position_test", catalog);
```
Comment (liurenjie1024) [inline icelake/tests/delete_test.rs:302]: Append module names in test name?

**Review by liurenjie1024 — COMMENTED** [review]
Generally LGTM, left some comments.

File: icelake/src/io/file_writer/position_delete_writer.rs (hunk 2 above)
Comment (ZENOTME) [reply icelake/src/io/file_writer/position_delete_writer.rs:23]: Roaring bitmap looks good to me. How about put it in a later refine PR.🤔

Comment (liurenjie1024) [reply icelake/src/io/file_writer/position_delete_writer.rs:23]: ok

File: icelake/src/io/position_delete_writer.rs
Comment (ZENOTME) [reply icelake/src/io/position_delete_writer.rs:1]: It has already been moved to file_writer.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**